semantics/src/main/scala/aqua/semantics/rules/ValuesAlgebra.scala

```diff
@@ -156,10 +158,16 @@ class ValuesAlgebra[S[_], Alg[_]: Monad](using
               }
               .map(_.sequence)
           )
+          empty <- ct.mode match {
+            // empty stream is not a Nil, it is an unique variable, that will be used as a stream
+            case CollectionToken.Mode.StreamMode =>
+              M.rename("stream-inline").map(n => VarRaw(n, StreamType(BottomType)))
```

Comment:
What about restricting this stream? Does this work inside `for`? Please write a test on that.

As I understand, this is a stream of bottom so that it is subtype of any stream. But it also means that inside function we can push to stream of bottom. It happens after type check so thats okay. But feels tricky. Maybe we should research into some kind of type guessing, e.g. pass `expectedType` to `valueToRaw` so that we could assign proper type here to an anon stream.

nit: I think `stream-anon` is better name maybe?